ecl/eclccserver/eclccserver.cpp

```diff
@@ -881,36 +918,71 @@ class EclccServer : public CInterface, implements IThreadFactory, implements IAb
     bool running;
     CSDSServerStatus serverstatus;
     Owned<IJobQueue> queue;
+    CriticalSection updateConfigCS;
+    StringAttr updatedQueueNames;
+    unsigned reloadConifgCBId = 0;
 
+
+    void configUpdate()
+    {
+        StringBuffer newQueueNames;
+        getQueues(newQueueNames);
+        if (!newQueueNames.length())
+            ERRLOG("No queues found to listen on");
+        if (!strsame(queueNames, newQueueNames))
+        {
+            updatedQueueNames.set(newQueueNames);
+            queue->cancelAcceptConversation();
+            PROGLOG("Updating queue due to queue names change from '%s' to '%s'", queueNames.str(), newQueueNames.str());
```

Comment:
This tracing also needs to be within the critical section

Reply:
yes, actually so does 'queue', but will link and and call cancelAcceptConversation outside.